.github/workflows/codeql.yml

```diff
@@ -0,0 +1,203 @@
+    #   - https://gh.io/recommended-hardware-resources-for-running-codeql
+    #   - https://gh.io/supported-runners-and-hardware-resources
+    #   - https://gh.io/using-larger-runners (GitHub.com only)
+    # Consider using larger runners or machines with greater resources for possible analysis time improvements.
+    runs-on: ${{ (matrix.language == 'swift' && 'macos-latest') || 'ubuntu-latest' }}
+    permissions:
+      # required for all workflows
+      security-events: write
+
+      # required to fetch internal or private CodeQL packs
+      packages: read
+
+      # only required for workflows in private repositories
+      actions: read
+      contents: read
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+        # - language: actions
+        #   build-mode: none
+        # - language: c-cpp
+        #   build-mode: autobuild
+        # - language: javascript-typescript
+        #   build-mode: none
+        - language: python
+          build-mode: none
+        # CodeQL supports the following values keywords for 'language': 'actions', 'c-cpp', 'csharp', 'go', 'java-kotlin', 'javascript-typescript', 'python', 'ruby', 'swift'
+        # Use `c-cpp` to analyze code written in C, C++ or both
+        # Use 'java-kotlin' to analyze code written in Java, Kotlin or both
+        # Use 'javascript-typescript' to analyze code written in JavaScript, TypeScript or both
+        # To learn more about changing the languages that are analyzed or customizing the build mode for your analysis,
+        # see https://docs.github.com/en/code-security/code-scanning/creating-an-advanced-setup-for-code-scanning/customizing-your-advanced-setup-for-code-scanning.
+        # If you are analyzing a compiled language, you can modify the 'build-mode' for that language to customize how
+        # your codebase is analyzed, see https://docs.github.com/en/code-security/code-scanning/creating-an-advanced-setup-for-code-scanning/codeql-code-scanning-for-compiled-languages
+    steps:
+    - name: Harden the runner (Audit all outbound calls)
+      uses: step-security/harden-runner@4d991eb9b905ef189e4c376166672c3f2f230481 # v2.11.0
+      with:
+        egress-policy: audit
+
+    - name: Checkout repository
+      uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      with:
+        fetch-depth: 2
+        submodules: recursive
+
+    # prevent running out of disk space on Ubuntu runners.
+    - name: Maximize build space
+      if: ${{ matrix.language == 'c-cpp' }}
+      uses: AdityaGarg8/remove-unwanted-software@90e01b21170618765a73370fcc3abbd1684a7793 # v5
+      with:
+        verbose: 'true'
+        remove-android: 'true'      # (frees ~9 GB)
+        remove-cached-tools: 'true' # (frees ~8.3 GB)
+
+    # Install FreeCAD dependencies (cpp)
+    - name: Setup build environment
+      if: ${{ matrix.language == 'c-cpp' }}
+      run: |
```

Comment:
Can you replace this with running `package/ubuntu/install-apt-packages.sh` so we only need maintain a single file for dependencies on Ubuntu?

Reply:
At the moment cpp scan is actually disabled, would it be ok to merge this now like this and I will make this change and test it before I make a new PR when I will also activate the cpp scan?